From cfc96b383c038b254af9c97ca84012f7d4709263 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sun, 6 Feb 2022 12:05:15 +0000
Subject: [PATCH] Correctly handle ipk URIs

These URIs should be tansformed to file paths before handling them locally

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Upstream-Status: Pending
---
 src/step/IpkInstallStep.cpp | 5 +++++
 src/step/IpkParseStep.cpp   | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/src/step/IpkInstallStep.cpp b/src/step/IpkInstallStep.cpp
index afca157..86d7802 100644
--- a/src/step/IpkInstallStep.cpp
+++ b/src/step/IpkInstallStep.cpp
@@ -56,6 +56,11 @@ bool IpkInstallStep::proceed(Task *task)
     bool verify = param["verify"].asBool();
     std::string ipkFile= param["ipkurl"].asString();
     bool allowDowngrade = param["allowDowngrade"].asBool();
+    
+    if( 0 == ipkFile.find("file://") ) {
+        // uri pointing to local file => transform to file path
+        ipkFile.replace(0,7,"");
+    }
 
     AppInstallerUtility::Result result =
             m_installerUtility.install(ipkFile, 0, verify, allowDowngrade, task->isAllowReInstall(), task->getInstallBasePath(),
diff --git a/src/step/IpkParseStep.cpp b/src/step/IpkParseStep.cpp
index 9c487bf..58a9360 100644
--- a/src/step/IpkParseStep.cpp
+++ b/src/step/IpkParseStep.cpp
@@ -40,6 +40,11 @@ bool IpkParseStep::proceed(Task *task)
     m_verify = param["verify"].asBool();
     m_appId = param["id"].asString();
     m_ipkFile = param["ipkurl"].asString();
+    
+    if( 0 == m_ipkFile.find("file://") ) {
+        // uri pointing to local file => transform to file path
+        m_ipkFile.replace(0,7,"");
+    }
 
     determineInstallPath();
 
